src/StreamrClient.js

```diff
@@ -778,7 +779,10 @@ export default class StreamrClient extends EventEmitter {
             const request = SubscribeRequest.create(sub.streamId, sub.streamPartition, sessionToken)
             debug('_requestSubscribe: subscribing client: %o', request)
             sp.setSubscribing(true)
-            this.connection.send(request)
+            await this.connection.send(request).catch((err) => {
+                sub.setState(Subscription.State.unsubscribed)
+                this.emit('error', `Failed to send subscribe request: ${err}`)
```

Comment:
Hm, should emit Error objects otherwise you lose stack trace.
```suggestion
                const error = new Error(`Failed to send subscribe request: ${err}`)
                error.stack = err.stack // this isn't ideal because it loses the above error's stack but not much you can do
                this.emit('error', err)
```

Reply:
Or do something like:

```suggestion
                debug(`Failed to send subscribe request: ${err}`)
                this.emit('error', err)
```
This might be better.